### PR TITLE
Allow DEK to be a CIP8 signer

### LINKED ISSUE #7459: CIP8 signatures by DEK should be valid
### Expected Behavior

DEK should be a valid signer

### Current Behavior

Only authorized signers on `Account.sol` are valid

### PR DESCRIPTION
**Pull request by nambrot**
### Description

This PR allows the DEK to be a CIP8 signer

### Other changes

- Set jest dependency resolution

### Tested

- Unit test 
- 
### Related issues

- Fixes #7459 

### Backwards compatibility

- Older clients might not recognize future clients using the DEK

### CODE REVIEW

**Review by medhakothari — APPROVED** [review]
LGTM

File: packages/sdk/identity/src/offchain-data-wrapper.ts [hunk 1]
```diff
@@ -173,12 +175,15 @@ class StorageRoot {
...
     if (await accounts.isAccount(this.account)) {
-      const signers = await Promise.all([
+      const keys = await Promise.all([
         accounts.getVoteSigner(this.account),
         accounts.getValidatorSigner(this.account),
         accounts.getAttestationSigner(this.account),
+        accounts.getDataEncryptionKey(this.account),
       ])
-      if (signers.some((signer) => signer === guessedSigner)) {
+      const signers = [keys[0], keys[1], keys[2], publicKeyToAddress(ensureUncompressed(keys[3]))]
```
Comment (codyborn) [inline packages/sdk/identity/src/offchain-data-wrapper.ts]: Can `getDataEncryptionKey()` return `undefined`? I see checks for it elsewhere in the code so we might want to add that check here before calling `ensureUncompressed()`

File: packages/sdk/identity/src/offchain-data-wrapper.test.ts [hunk 2]
```diff
@@ -142,6 +143,50 @@ testWithGanache('Offchain Data', (web3) => {
...
+      const DEKAddress = privateKeyToAddress(writerDEK)
+      compressedWriter.wrapper.kit.connection.addAccount(writerDEK)
+      compressedWriter.wrapper.signer = DEKAddress
+
+      const nameAccessor = new PublicNameAccessor(compressedWriter.wrapper)
+      await nameAccessor.write(testPayload)
+
+      const readerNameAccessor = new PublicNameAccessor(reader.wrapper)
+      const resp = await readerNameAccessor.readAsResult(compressedWriter.address)
+      if (resp.ok) {
```
Comment (codyborn) [inline packages/sdk/identity/src/offchain-data-wrapper.test.ts]: nitpick: This is repeated in multiple test cases so we can create a common method

**Review by codyborn — APPROVED** [review]
Left some small feedback. Thanks!

**Pull request merged**